tests/test_backend.py

```diff
@@ -417,34 +418,82 @@ def mock_open_run(*_, data=None):
 
     # Reprocess with `data='all'`, but as if there is no proc data
     with patch("ctxrunner.extra_data.open_run", side_effect=mock_open_run):
-        main(['1234', '42', 'all', '--save', str(out_path)])
+        main(['exec', '1234', '42', 'all', '--save', str(out_path)])
 
     # Check that `meta_array` wasn't processed, since it requires proc data
     with h5py.File(out_path) as f:
         assert "meta_array" not in f
 
     # Reprocess with proc data
     with patch("ctxrunner.extra_data.open_run", return_value=mock_run):
-        main(['1234', '42', 'all', '--save', str(out_path)])
+        main(['exec', '1234', '42', 'all', '--save', str(out_path)])
 
     # Now `meta_array` should have been processed
     with h5py.File(out_path) as f:
         assert "meta_array" in f
 
     # Runs shouldn't be opened with open_run() when --mock is passed
     with patch("ctxrunner.extra_data.open_run") as open_run:
-        main(["1234", "42", "all", "--mock"])
+        main(["exec", "1234", "42", "all", "--mock"])
         open_run.assert_not_called()
 
     # When only proc variables are reprocessed, the run should still be opened
     # with `data='all'` so that raw data is available. Note that we patch
     # Results too so that the test doesn't throw exceptions.
     with patch("ctxrunner.extra_data.open_run") as open_run, \
          patch("ctxrunner.Results"):
-        main(["1234", "42", "proc"])
+        main(["exec", "1234", "42", "proc"])
 
         open_run.assert_called_with(1234, 42, data="all")
 
+def test_custom_environment(mock_db, virtualenv, monkeypatch, qtbot):
+    db_dir, db = mock_db
+    monkeypatch.chdir(db_dir)
+
+    ctxrunner_deps = ["extra_data"]
+
+    # Install dependencies for ctxrunner and a light-weight package (sfollow)
+    # that isn't in our current environment.
+    virtualenv.install_package(" ".join([*ctxrunner_deps, "sfollow"]),
+                               installer="pip install")
+
+    # Write a context file that requires the new package
+    new_env_code = f"""
+    from pathlib import Path
+
+    import sfollow
+    from damnit_ctx import Variable
+
+    # Ensure that the context file is *always* evaluated in the database
+    # directory. This is necessary to ensure that things like relative imports
+    # and janky sys.path shenanigans work.
+    assert str(Path.cwd()) == "{db_dir}"
```

Comment:
```suggestion
    assert str(Path.cwd()) == {str(db_dir)!r}
```

The `!r` should correctly manage any quotes or weird characters to produce a string literal. Not that I think there's likely to be anything weird in the path, but it's easy to do it right. :slightly_smiling_face:

Reply:
Fixed in 2a041c8.